### PR TITLE
gh-125442: Use _PyLong_GetOne() and _PyLong_GetZero() in Objects

### PR DESCRIPTION
**Pull request by hikariyo**
Simply reduces a null check as `_PyLong_GetOne()` cannot fail.

* Issue: gh-125442

### CODE REVIEW

**Comment by skirpichev** [comment]
There are other places, that might be changed, especially in longobject.c:
```
$ git grep 'PyLong_From[IL].*([10]' Objects/|wc -l
22
```
I suggest you first create an issue.

File: Objects/iterobject.c
Comment (hikariyo) [inline Objects/iterobject.c:1]: Sorry for typo in commit message🥲
It's pycore_long.h

**Review by serhiy-storchaka — CHANGES_REQUESTED** [review]
Do not merge until there are solid evidences of the speed up.

**Comment by skirpichev** [comment]
@serhiy-storchaka, why do you suspect speed regressions at all?  PyLong_FromLong() will run more complex versions of _PyLong_GetZero/One anyway.

https://github.com/python/cpython/blob/5dac0dceda9097d46a0b5a6ad7c927e002c6c7a5/Objects/longobject.c#L309-L317

https://github.com/python/cpython/blob/5dac0dceda9097d46a0b5a6ad7c927e002c6c7a5/Objects/longobject.c#L56-L61

https://github.com/python/cpython/blob/5dac0dceda9097d46a0b5a6ad7c927e002c6c7a5/Include/internal/pycore_long.h#L66-L74

**Comment by hikariyo** [comment]
@serhiy-storchaka I'm inspired by #125044 and I found that there are other places using `PyLong_FromLong` for 0 or 1.

At least 2 null checks below was removed after these modifications.

https://github.com/python/cpython/blob/5dac0dceda9097d46a0b5a6ad7c927e002c6c7a5/Objects/floatobject.c#L1564-L1566

https://github.com/python/cpython/blob/5dac0dceda9097d46a0b5a6ad7c927e002c6c7a5/Objects/longobject.c#L5006-L5008

**Comment by skirpichev** [comment]
@hikariyo, all similar checks are unreachable now.  But that's might be viewed as internal detail of the PyLong_FromLong() implementation.

**Comment by serhiy-storchaka** [comment]
I suspect no difference at all. This is a pure cosmetic change. On other hand, it replaces public functions calls with private functions calls. So, this is a cosmetic change that makes the code more ugly.

**Comment by hikariyo** [comment]
> @hikariyo, all similar checks are unreachable now. But that's might be viewed as internal detail of te PyLong_FromLong() implementation.

However _PyLong_GetZero/One must return non-null values, so using them can avoid this problem.

**Comment by hikariyo** [comment]
> I suspect no difference at all. This is a pure cosmetic change. On other hand, it replaces public functions calls with private functions calls. So, this is a cosmetic change that makes the code more ugly.

But _PyLong_GetZero/One do widely used in current code. Shouldn't we unify the usages?

**Comment by serhiy-storchaka** [comment]
They were introduced (first as simple global variables) for the cases when you need a borrowed reference to integers 0 or 1. `PyLong_FromLong()` continued to be used in cases when you need a new reference.

Closing this PR as pure cosmetic change.

**Pull request closed without merging**